spec/features/peeps_management_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'spec_helper'
+require_relative 'helpers/session'
+require_relative 'helpers/peeps'
+
+include SessionHelpers
+include PeepsHelpers
+
+feature 'User posts a peep' do
+
+  before(:each) do
+    User.create(email: 'uhoh@danieljohnston.co.uk',
+                handle: 'hrrmm',
+                password: 'testpassword',
+                password_confirmation: 'testpassword')
+  end
+
+  scenario 'and receives confirmation of posting' do
+    sign_in
+    post_peep
+    expect(page).to have_content('Your peep has been posted')
+  end
+
+  scenario 'while not logged in' do
+    expect { post_peep }.to raise_error('Unable to find field :content')
+    expect(page).to have_content('You need to be logged in first')
+  end
+
+  scenario 'and sees their peep on the front page' do
+    sign_in
+    post_peep
+    visit '/'
+    expect(page).to have_content('This is a test peep')
+  end
+end
+
+feature 'User views list of peeps' do
```

Comment:
I had a go at refactoring this: https://gist.github.com/ptolemybarnes/323b9559049264dc6c2d
Again, not convinced this is the best way of going about testing that peeps appear in the order you want but it's a little cleaner.